packages/viewport/src/with-viewport-match.js

```diff
@@ -1,13 +1,83 @@
 /**
  * External dependencies
  */
-import { mapValues } from 'lodash';
+import { mapValues, reduce } from 'lodash';
 
 /**
  * WordPress dependencies
  */
+import { createContext, useContext, useMemo } from '@wordpress/element';
 import { createHigherOrderComponent } from '@wordpress/compose';
-import { withSelect } from '@wordpress/data';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Hash of breakpoint names with pixel width at which it becomes effective.
+ *
+ * @see _breakpoints.scss
+ *
+ * @type {Object}
+ */
+export const BREAKPOINTS = {
+	huge: 1440,
+	wide: 1280,
+	large: 960,
+	medium: 782,
+	small: 600,
+	mobile: 480,
+};
+
+/**
+ * Hash of query operators with corresponding condition for media query.
+ *
+ * @type {Object}
+ */
+export const OPERATORS = {
+	'<': 'max-width',
+	'>=': 'min-width',
+};
+
+const OPERATOR_EVALUATORS = {
+	'<': ( breakpointValue, width ) => ( width < breakpointValue ),
+	'>=': ( breakpointValue, width ) => ( width >= breakpointValue ),
+};
+
+const ViewportMatchWidthContext = createContext( null );
+
+/**
+ * Component that makes all withViewportMatch usages descendents of its children,
+ * evaluate as if the viewport had the width specific as a prop.
+ */
+export const ViewportMatchWidthProvider = ( { width, children } ) => {
+	const queriesResult = useMemo(
```

Comment:
What is a "queries result" (shape, purpose)? I'm sure I could follow it along, but my initial impression is it's unclear what we're computing. An inline comment or JSDoc could help clarify.

Reply:
I almost wonder if it's even worth computing this cached value ahead of time. There are so few `withViewportMatch` (likely often zero within a provider?) and the operation seems reasonably trivial to compute on the fly within `withViewportMatch` than to compute all possible options ahead of time. Also keeps things a little easier to follow for that logic to be colocated.